### PR TITLE
650 randomised journal list

### LINKED ISSUE #650: Make randomizable list of journals in Huddersfield-style
Hudds press looks great. We need to build the grid of journals into a randomizable format.

### PR DESCRIPTION
**Pull request by ajrbyers**
Closes #650

### CODE REVIEW

**Comment by mauromsl** [comment]
Seems like this one wasn't branched from master

**Comment by ajrbyers** [comment]
> Seems like this one wasn't branched from master

Yeah looks like 593-journal_code_field

File: src/core/homepage_elements/journals/views.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def featured_journals(request):
                 )
             )
 
-    template = 'featured_journals.html'
```
Comment (ajrbyers) [inline src/core/homepage_elements/journals/views.py]: I think if this is reverted the rest of the PR is fine.

**Comment by ajrbyers** [comment]
Ok, conflict here also resolved.

File: src/core/plugin_loader.py [hunk 2]
```diff
@@ -38,25 +41,28 @@ def load(directory="plugins", prefix="plugins", permissive=False):
...
+                    STAGE_CHOICES.append(
+                        (plugin_settings.STAGE, plugin_settings.PLUGIN_NAME)
+                    )
+                    ELEMENT_STAGES[
+                        plugin_settings.PLUGIN_NAME] = [plugin_settings.STAGE]
+
+                # Call event registry
+                register_for_events(plugin_settings)
+            except ModuleNotFoundError as e:
+                logger.error('Could not lod module, error: {e}'.format(e=e))
```
Comment (mauromsl) [inline src/core/plugin_loader.py]: This seems to be from another branch

File: src/static/OLH/css/app.css [hunk 3]
```diff
@@ -5250,3 +5253,11 @@ button.list-group-item-danger.active:focus {
...
 .feature-article-image {
   object-fit: cover; }
+
+.alignright {
+  float: right;
+  padding: 5px; }
+
+.alignleft {
+  float: left;
+  padding: 5px; }
```
Comment (mauromsl) [inline src/static/OLH/css/app.css]: This seems to be from another branch. It is not even used

File: src/static/OLH/css/app.css [hunk 4]
```diff
@@ -4650,6 +4650,9 @@ main section {
   main section.no-padding {
     padding: 0; }
 
+#article {
+  word-wrap: break-word; }
+
```
Comment (mauromsl) [inline src/static/OLH/css/app.css]: This seems to be from another branch

File: src/utils/management/commands/install_janeway.py [hunk 5]
```diff
@@ -57,7 +57,17 @@ def handle(self, *args, **options):
...
+            while True:
+                code = input('Journal #1 code (15 characters max): ')
+
+                if len(code) <= 15:
+                    journal.code = code
+                    journal.save()
+                    break
+                else:
+                    print('Journal code must be 15 characters or less.')
+
```
Comment (mauromsl) [inline src/utils/management/commands/install_janeway.py]: This seems to be from another branch

File: src/core/homepage_elements/journals_and_html/forms.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+from django import forms
+
+from press import models
+from journal import models as jm
+from django_summernote.widgets import SummernoteWidget
+
+
+class FeaturedJournalsForm(forms.ModelForm):
```
Comment (mauromsl) [inline src/core/homepage_elements/journals_and_html/forms.py]: There is a `FeaturedJournalsForm` in `src/core/homepage_elements/journals/forms.py` already, which also controls `random_featured_journals` and controls `models.Press`

File: src/core/plugin_loader.py (hunk 2 above)
Comment (ajrbyers) [reply src/core/plugin_loader.py]: I'm not sure if this is from another branch as I did do some work on handling plugin loading errors for HPEs.

**Pull request closed without merging**